scripts/dts/gen_defines.py

```diff
@@ -771,6 +776,17 @@ def out(ident, val, aliases=()):
     return primary_ident
 
 
+def out_define(ident, val, deprecation_msg, out_file):
+    # out() helper for writing a #define. See out_dev() for the meaning of
+    # 'deprecation_msg'.
+
+    s = "#define DT_{:40}".format(ident)
+    if deprecation_msg:
+        s += r' _Pragma("GCC warning \"{}\"")'.format(deprecation_msg)
```

Comment:
Can we have this be __DEPRECATED_MACRO instead.  So that way this isn't GCC specific.